### DISCUSSION TITLE
Linear regression example - why need to unsqueeze X?

### DISCUSSION THREAD

**Discussion by vinclei** [mrdbourke/pytorch-deep-learning#316] [Q&A] [answered]
In chapter "35 Creating a dataset with linear regression", we generated X and unsqueeze it: 
`X = torch.arange(start, end, step).unsqueeze(dim=1)`
It added another dimension to X, why do we need that? I tried without `unsqueeze` and seems the workflow is still working.

**Comment by vinclei** [comment, marked as answer]
Update for future reference: 
It's fine not `unsqueeze` if we set up parameters using `nn.Parameter()`, but it will through shape error if using `nn.Linear()`
